pallets/task/src/lib.rs

```diff
@@ -333,6 +380,34 @@ pub mod pallet {
 			Ok(task_id)
 		}
 
+		// Task can be updated only after it has been created. Task that is already in progress can't be updated.
+		pub fn update_created_task(from_initiator: &T::AccountId, task_id: &T::Hash, new_title: &[u8], new_specification: &[u8], new_budget: &BalanceOf<T>, new_deadline: u64) -> Result<(), DispatchError> {
+
+			// Check if task exists
+			let mut task = Self::tasks(&task_id).ok_or(<Error<T>>::TaskNotExist)?;
+
+			// Check if the owner is the one who created task
+			ensure!(Self::is_task_initiator(task_id, from_initiator)?, <Error<T>>::OnlyInitiatorUpdatesTask);
+
+			// Ensure user has a profile before creating a task
+			ensure!(pallet_profile::Pallet::<T>::has_profile(from_initiator).unwrap(), <Error<T>>::NoProfile);
+			
+			// Ensure deadline is in the future
+			let deadline_duration = Duration::from_millis(task.deadline.saturated_into::<u64>());
+			ensure!(T::Time::now() < deadline_duration, Error::<T>::IncorrectDeadlineTimestamp);
+
```

Comment:
shouldn't it check that task status is not == progress ?

Reply:
or better check status is == Created

Reply:
Good catch! I will add the check.